agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/ResolveMarchOrderGameState.ts

```diff
@@ -18,8 +18,13 @@ import TakeControlOfEnemyPortGameState, { SerializedTakeControlOfEnemyPortGameSt
 import { findOrphanedShipsAndDestroyThem } from "../../port-helper/PortHelper";
 
 export default class ResolveMarchOrderGameState extends GameState<ActionGameState, ResolveSingleMarchOrderGameState | CombatGameState | TakeControlOfEnemyPortGameState> {
+    public currentTurnOrderIndex: number;
```

Comment:
One note: I guess you have to add this field to `SerizalizedMarchOrderGameState`. See e.g. `ResolveSingleMarchOrderGameState` for an example with a GameState that holds fields. This is necessary to recover the state when the server loads the game from the DB.

Reply:
If we change how its serialized/deserialized to the DB, wouldn't that require some special attention to the current live games (since they'll adhere to the old format)?

Reply:
Yep... Long will have to create the necessary migration for this change.

Reply:
There's a migration system for this kind of processing. You can see quickly how it works in `src/server/serializedGameMigrations.ts`.

It basically takes a serializedGame of the previous format version, and apply the necessary transformation on it to make it compatible the new format of serialization.

Once it's review, I can merge this and make the migration in an other PR.